### DISCUSSION TITLE
uploadFile -> Error: $ operator is invalid for atomic vectors

### DISCUSSION THREAD

**Discussion by bonushenricus** [eblondel/zen4R#80] [Q&A] [answered]
Hi Emmanuel
now I have this problem
after the deposit
I have to upload a file before publishing
but this
`zenodo$uploadFile('/home/bonushenricus/Documenti/lavoro/progetti/vindicta/kobo/foto_ovature/ENGA22060401.jpeg', myrec$id)`
give me this error

> Error: $ operator is invalid for atomic vectors

what am I still doing wrong?

**Comment by eblondel** [comment, marked as answer]
Hi @bonushenricus i've converted this in discussion (feel free to open one for questions, we use more issues for enhancements/new features or bug fixing), although here the documentation has to be updated here. In the past the `id` had to be used. It can still be used with the `recordId` argument of `uploadFile` method, but in that case the old upload API will be used. 
Please use the `record` argument, which will enable the new upload API (bucket-based), in that way:

`zenodo$uploadFile('/yourfile', record = myrec)`

**Comment by bonushenricus** [comment]
Thanky you. It works!
